### PR TITLE
Reland: [Impeller] Use downsample shader for blur instead of mip levels.

### PR DESCRIPTION
**Pull request by gaaclarke**
relands https://github.com/flutter/engine/pull/53760
reverted in https://github.com/flutter/engine/pull/54148

The fix for this was found with the help of Jonah.

Part of a series of gaussian blur changes:
1) https://github.com/flutter/engine/pull/54148
1) https://github.com/flutter/engine/pull/54116
1) https://github.com/flutter/engine/pull/54150
1) https://github.com/flutter/engine/pull/54149

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: impeller/entity/contents/filters/gaussian_blur_filter_contents.cc [hunk 1]
```diff
@@ -360,12 +360,15 @@ fml::StatusOr<RenderTarget> MakeDownsampleSubpass(
...
                                 command_buffer, subpass_callback);
   } else {
-    // This assumes we don't scale below 1/8
+    // This assumes we don't scale below 1/16.
     Scalar edge = 1.0;
     Scalar ratio = 0.25;
-    if (pass_args.effective_scalar.x <= 0.125f) {
+    if (pass_args.effective_scalar.x <= 0.0625f) {
+      edge = 7.0;
+      ratio = 1.0f / 64.0f;
```
Comment (jonahwilliams) [inline impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:368]: probably a better way of doing it...

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by gaaclarke** [comment]
This PR is counting on https://github.com/flutter/engine/pull/54150 to get it below the 1.0 threshold.  That PR however creates some artifacts at the edges of 1D blurs.  I'll look to see if we can fix that otherwise we can look into other options.

**Comment by jonahwilliams** [comment]
Does this still visually look wrong even if the threshold is above 1?

**Comment by gaaclarke** [comment]
I just forgot to push my fix for the 1d blur and rebase.  I think we are good.

**Comment by gaaclarke** [comment]
Or more accurately, once https://github.com/flutter/engine/pull/54150 lands this will pass.

**Comment by gaaclarke** [comment]
This PR coops old tests but maintains an old branch in the code for when mipmaps are used.  I verified that the following tests still exercise that old branch.  We might want something more apropos though to make sure that we don't get rid of testing that branch.

```
Play/AiksTest.BlurredRectangleWithShader/Metal
Play/AiksTest.MaskBlurVariantTestSolidTranslucentWithFilters/Metal
Play/AiksTest.MaskBlurVariantTestInnerTranslucentWithBlurImageFilter/Metal
Play/AiksTest.MaskBlurVariantTestOuterOpaqueWithBlurImageFilter/Metal
Play/AiksTest.GaussianBlurSolidColorTinyMipMap/Metal
Play/AiksTest.GaussianBlurBackdropTinyMipMap/Metal
Play/AiksTest.CanRenderEmojiTextFrameWithBlur/Metal
Play/DlGoldenTest.TextBlurMaskFilterDisrespectCTM/Metal
Play/DlGoldenTest.GaussianVsRRectBlurScaled/Metal
Play/DlGoldenTest.GaussianVsRRectBlurScaledRotated/Metal
Play/DlGoldenTest.FastVsGeneralGaussianMaskBlur/Metal
```

**Pull request merged**